src/test/scala/sigmastate/lang/SigmaParserTest.scala

```diff
@@ -531,18 +532,179 @@ class SigmaParserTest extends PropSpec with PropertyChecks with Matchers with La
+  }
+
+  property("SNumeric.toBits") {
+    parse("1.toBits") shouldBe Select(IntConstant(1), "toBits")
+    parse("1L.toBits") shouldBe Select(LongConstant(1), "toBits")
+  }
+
+  property("SNumeric.abs") {
+    parse("1.abs") shouldBe Select(IntConstant(1), "abs")
+  }
+
+  property("SNumeric.compare") {
+    parse("1.compare(2)") shouldBe Apply(Select(IntConstant(1), "compare", None), Vector(IntConstant(2)))
+  }
+
+  property("numeric constant negation") {
+    parse("-3") shouldBe IntConstant(-3)
+    parse("-3.toByte") shouldBe Select(IntConstant(-3), "toByte")
+  }
+
+  property("numeric negation unary op") {
+    parse("-OUTPUTS.size") shouldBe Negation(Select(Ident("OUTPUTS"), "size").asIntValue)
+    parse("- (3 - 2)") shouldBe Negation(Minus(IntConstant(3), IntConstant(2)))
+  }
+
+  property("bitwise inversion unary op ~") {
+    parse("~OUTPUTS.size") shouldBe BitInversion(Select(Ident("OUTPUTS"), "size").asIntValue)
+  }
+
+  property("HEADERS methods") {
+    parse("HEADERS") shouldBe Ident("HEADERS")
+    parse("HEADERS.version") shouldBe Select(Ident("HEADERS"), "version")
+    parse("HEADERS.parentId") shouldBe Select(Ident("HEADERS"), "parentId")
+  }
+
+  property("logical not unary op") {
+    parse("!true") shouldBe LogicalNot(TrueLeaf)
+    parse("! (1 == 0)") shouldBe LogicalNot(EQ(IntConstant(1), IntConstant(0)))
+  }
+
+  property("logical XOR") {
+    parse("true ^ false") shouldBe MethodCallLike(TrueLeaf, "^", Vector(FalseLeaf))
+  }
+
+  property("BitAnd: bitwise AND for numeric types") {
+    parse("1 & 2") shouldBe BitOp(IntConstant(1), IntConstant(2), OpCodes.BitAndCode)
+  }
+
+  property("BitShiftRight: bit-shifted right for numeric types") {
+    parse("128 >> 2") shouldBe MethodCallLike(IntConstant(128), ">>", Vector(IntConstant(2)))
+  }
+
+  property("BitShiftLeft: bit-shifted left for numeric types") {
+    parse("128 << 2") shouldBe MethodCallLike(IntConstant(128), "<<",  Vector(IntConstant(2)))
+  }
+
+  property("BitShiftRightZeroed: bit-shifted right(zeroed) for numeric types") {
+    parse("128 >>> 2") shouldBe MethodCallLike(IntConstant(128), ">>>", Vector(IntConstant(2)))
+  }
+
+  property("BitCollShiftRight: bit-shifted right for numeric types") {
+    parse("Coll(true, false) >> 2") shouldBe
+      MethodCallLike(Apply(Ident("Coll"), Vector(TrueLeaf, FalseLeaf)), ">>", Vector(IntConstant(2)))
+  }
+
+  property("BitCollShiftLeft: bit-shifted left for numeric types") {
+    parse("Coll(true, false) << 2") shouldBe
+      MethodCallLike(Apply(Ident("Coll"), Vector(TrueLeaf, FalseLeaf)), "<<", Vector(IntConstant(2)))
+  }
+
+  property("BitCollShiftRightZeroed: bit-shifted right(zeroed) for numeric types") {
+    parse("Coll(true, false) >>> 2") shouldBe
+      MethodCallLike(Apply(Ident("Coll"), Vector(TrueLeaf, FalseLeaf)), ">>>", Vector(IntConstant(2)))
+  }
+
+  property("rotateLeft op: shift bit collection left") {
+    parse("Coll(true, false) rotateLeft 2") shouldBe
```

Comment:
@greenhat this should be method `rotateLeft` ( and the same for right)

Reply:
Missed that one. This makes so much more sense than binary operation. :) 
Fixed.